x-pack/legacy/plugins/maps/public/shared/components/metric_editor.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiFieldText,
+  EuiFormRow,
+} from '@elastic/eui';
+
+import {
+  MetricSelect,
+  METRIC_AGGREGATION_VALUES,
+} from './metric_select';
+import { SingleFieldSelect } from './single_field_select';
+
+export function MetricEditor({ fields, metricsFilter, metric, onChange }) {
+
+  const onAggChange = (metricAggregationType) => {
+    onChange({
+      ...metric,
+      type: metricAggregationType,
+    });
+  };
+  const onFieldChange = (fieldName) => {
+    onChange({
+      ...metric,
+      field: fieldName,
+    });
+  };
+  const onLabelChange = (e) => {
+    onChange({
+      ...metric,
+      label: e.target.value,
+    });
+  };
+
+  let fieldSelect;
+  if (metric.type && metric.type !== 'count') {
+    const filterNumberFields = (field) => {
+      return field.type === 'number';
+    };
+    fieldSelect = (
+      <EuiFormRow
+        label={i18n.translate('xpack.maps.metricsEditor.selectFieldLabel', {
+          defaultMessage: 'Field'
+        })}
+      >
+        <SingleFieldSelect
+          placeholder={i18n.translate('xpack.maps.metricsEditor.selectFieldPlaceholder', {
+            defaultMessage: 'Select field'
+          })}
+          value={metric.field}
+          onChange={onFieldChange}
+          filterField={filterNumberFields}
+          fields={fields}
+          isClearable={false}
+        />
+      </EuiFormRow>
+    );
+  }
+
+  let labelInput;
+  if (metric.type) {
+    labelInput = (
+      <EuiFormRow
+        label={i18n.translate('xpack.maps.metricsEditor.customLabel', {
+          defaultMessage: 'Custom label'
+        })}
+      >
+        <EuiFieldText
+          onChange={onLabelChange}
+          value={metric.label ? metric.label : ''}
```

Comment:
I would not display the custom label in the configuration this way. it "hides" a little what the actual term query-configuration is.

If we want to show it, I would do something like `sum bytes as  'my custom label'` or just leave it out altogether